src/jenkins_job_insight/main.py

```
@@ -2383,17 +2484,136 @@ async def api_dashboard() -> list[dict]:
+    settings: Settings = Depends(get_settings),
+) -> dict:
+    """Validate a GitHub or Jira token by making a lightweight API call.
+
+    GitHub: GET /user (returns authenticated user info)
+    Jira: GET /rest/api/2/myself (returns authenticated user info)
+    """
+    token = body.token.strip()
+    if not token:
+        return {"valid": False, "username": "", "message": "Token is required"}
+
+    if body.token_type == "github":
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to: "token_type"

(S105)